### ISSUE TITLE
azurerm_container_group image_registry_credential supports managed identity

### ISSUE THREAD

**Issue by kgopi1** [hashicorp/terraform-provider-azurerm#15915]
### Community Note

* Please vote on this issue by adding a 👍 [reaction](https://blog.github.com/2016-03-10-add-reactions-to-pull-requests-issues-and-comments/) to the original issue to help the community and maintainers prioritize this request
* Please do not leave "+1" or "me too" comments, they generate extra noise for issue followers and do not help prioritize the request
* If you are interested in working on this issue or have submitted a pull request, please leave a comment

### New or Affected Resource(s)

* azurerm_container_group

### Potential Terraform Configuration

Currently azurerm_container_group  image_registry_credential  supports only the user name and password , howerver ARM template allows to use managed identity. 

Can you add the support for identity ? 

Reference Link. 

https://docs.microsoft.com/en-us/azure/container-instances/using-azure-container-registry-mi

"imageRegistryCredentials": [
        {
            "server":"myacr.azurecr.io",
            "identity":"/subscriptions/xxxxxxxx-xxxx-xxxx-xxxx-xxxxxxxxxxxxx/resourcegroups/myResourceGroup/providers/Microsoft.ManagedIdentity/userAssignedIdentities/myACRId"
        }

```hcl
# Copy-paste your Terraform configurations here - for large Terraform configs,
# please use a service like Dropbox and share a link to the ZIP file. For
# security, you can also encrypt the files using our GPG public key.
```

### References

* #0000

**Comment by fsrajer** [comment]
I just wanted to request this myself. I will at least explain my experience.

It should be possible to use containers from `azurerm_container_registry` in `azurerm_container_group` if you set correctly `AcrPull` role assignment to the identity of the container group. The problem is however that this is not enough. The identity has to be defined as the method for authenticating in `image_registry_credential` (which is not implemented now).

There is a workaround possible, which is not very elegant but works. You can use `azurerm_resource_group_template_deployment` to define the resource, where you can use field `imageRegistryCredentials`. Here is a reference to full template definition: https://docs.microsoft.com/en-us/azure/templates/microsoft.containerinstance/containergroups?tabs=json

**Comment by speak2beeb** [comment]
Our org would love this feature as well!

**Comment by arman9249** [comment]
This will help me a lot. Are you considering to add this feature?

**Comment by javiervelamindcurv** [comment]
any update?

**Comment by KJaunay** [comment]
As it stands currently this feature is what is blocking the use of azure container instances and azure container registries with private networking at a fortune 500 company.

**Comment by jfair9** [comment]
This feature is also block my organization from furthering their support for ACI. Any updates on this feature?

**Comment by timosur** [comment]
This also blocks us currently from using ACIs in my org. Any update?

**Comment by isaac-collins** [comment]
Any update on this feature? My org is also blocked.

**Comment by KJaunay** [comment]
Possibly fixed by #17785? I can see the image_registry_credential attribute is supported in the latest (v3.30.0) of the provider (link)[https://registry.terraform.io/providers/hashicorp/azurerm/latest/docs/resources/container_group#image_registry_credential]

**Comment by rcskosir** [comment]
Thanks for taking the time to submit this issue. It looks like this has been resolved as of #17785. As such, I am going to mark this issue as closed.
